### PR TITLE
monitor: add tests

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

/kind api-change
/kind cleanup
/kind failing-test

**What this PR does / why we need it**:
- Add more tests
- Use prometheus/client_go to make life easy
- Upgrade API version to v1alpha3 due to break changes https://github.com/kubesphere/kubesphere/pull/1751

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 	configv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/config/v1alpha2"
 	iamv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/iam/v1alpha2"
 	loggingv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/logging/v1alpha2"
-	monitoringv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/monitoring/v1alpha2"
+	monitoringv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/monitoring/v1alpha3"
```
Comment (benjaminhuo) [inline pkg/apiserver/apiserver.go:31]: monitoringv1alpha2 doesn't match "kubesphere.io/kubesphere/pkg/kapis/monitoring/v1alpha3"
if it's indeed v1alpha3, maybe we can change it to something like v1beta1 ?

File: tools/cmd/doc-gen/main.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ import (
 	devopsv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/devops/v1alpha2"
 	iamv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/iam/v1alpha2"
 	loggingv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/logging/v1alpha2"
-	monitoringv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/monitoring/v1alpha2"
+	monitoringv1alpha2 "kubesphere.io/kubesphere/pkg/kapis/monitoring/v1alpha3"
```
Comment (benjaminhuo) [inline tools/cmd/doc-gen/main.go:35]: The same problem here: monitoringv1alpha2  doesn't match "kubesphere.io/kubesphere/pkg/kapis/monitoring/v1alpha3"

**Comment by benjaminhuo** [comment]
It's great we are adding more unit test code, just one minor problem I found

File: pkg/apiserver/apiserver.go (hunk 1 above)
Comment (huanggze) [reply pkg/apiserver/apiserver.go]: As the monitoring api group is growing, I think we are still at the alpha stage. For example new apis for custom monitoring will come in soon.

File: tools/cmd/doc-gen/main.go (hunk 2 above)
Comment (huanggze) [reply tools/cmd/doc-gen/main.go]: done

File: pkg/apiserver/apiserver.go (hunk 1 above)
Comment (huanggze) [reply pkg/apiserver/apiserver.go]: done

**Comment by zryfish** [comment]
The pr looks good overall, but it's lack of documentation, please add more comments to clarify. Also, please resolve the conflicts

**Comment by huanggze** [comment]
done @zryfish

**Pull request merged**